### PR TITLE
[INT-1439] add support to import database dump

### PR DESCRIPTION
**Pull request by tinect**

### CODE REVIEW

File: devenv.nix [hunk 1]
```diff
@@ -286,4 +287,55 @@ in {
...
+    echo "Are you sure you want to download the file and overwritting database shopware with its data (y/n)?"
+    read answer
+
+    if [[ "$answer" != "y" ]]; then
+        echo "Allright, we will stop here."
+        exit
+    fi
+
+    BASENAME=$(basename -- ${config.env.DATABASE_DUMP_URL})
+    TARGETFOLDER="${config.env.DEVENV_STATE}/importdatabase_dump"
```
Comment (jochenmanz) [inline devenv.nix:306]: "importdb" as its the name as the script name-

File: devenv.nix [hunk 2]
```diff
@@ -286,4 +287,55 @@ in {
...
+        curl --create-dirs "${config.env.DATABASE_DUMP_URL}" --output "$TARGETFOLDER/latest.sql.zip"
+        unzip -j -o "$TARGETFOLDER/latest.sql.zip" '*.sql' -d "$TARGETFOLDER"
+    else
+        echo "unsupported file type"
+        exit
+    fi
+
+    rm -f "$TARGETFOLDER/latest.sql.zip"
+
+    SQLFILE=$(find "$TARGETFOLDER" -name "*.sql" | head -n 1)
```
Comment (jochenmanz) [inline devenv.nix:327]: SQL_FILE? :)

File: devenv.nix [hunk 3]
```diff
@@ -286,4 +287,55 @@ in {
...
+
+    echo "Are you sure you want to download the file and overwritting database shopware with its data (y/n)?"
+    read answer
+
+    if [[ "$answer" != "y" ]]; then
+        echo "Allright, we will stop here."
+        exit
+    fi
+
+    BASENAME=$(basename -- ${config.env.DATABASE_DUMP_URL})
```
Comment (jochenmanz) [inline devenv.nix:305]: do you need a basename when analyzing the used extension? if not this can just be removed :)

File: devenv.nix [hunk 4]
```diff
@@ -286,4 +287,55 @@ in {
   scripts.debug.exec = ''
     XDEBUG_SESSION=1 ${phpXdebug}/bin/php "$@"
   '';
+
+  scripts.importdb.exec = ''
```
Comment (jochenmanz) [inline devenv.nix:291]: could you move the content to the let part of this file?

**Comment by jochenmanz** [comment]
@tinect could you add an example to the readme.md?

File: devenv.nix (hunk 1 above)
Comment (tinect) [reply devenv.nix:306]: makes sense 👍

File: devenv.nix (hunk 2 above)
Comment (tinect) [reply devenv.nix:327]: sounds good

File: devenv.nix (hunk 3 above)
Comment (tinect) [reply devenv.nix:305]: Oh yeah, we can save this variable. Will be removed

File: devenv.nix (hunk 4 above)
Comment (tinect) [reply devenv.nix]: 👍 Sure! changed!

**Comment by Songworks** [comment]
Can it be possible to import multiple sql files too?

For example:

* `sql_dumps/01_latest_dump.sql.zip` => Anonymized dump downloaded from remote live/stage
* `sql_dumps/20_admins.sql` => Inserts common admin users
* `sql_dumps/30_domains.sql` => Inserts local hosts/domains into the sales channel(s)

Last one sounded useful to me while writing this.
Cheers!

File: devenv.nix [hunk 5]
```diff
@@ -71,13 +71,17 @@ let
     '') entries}
 
     # default config
-    ${updateConfig} core.mailerSettings.emailAgent ""
+    ${scriptResetEmailAgent}
```
Comment (jochenmanz) [inline devenv.nix:74]: lets move all config changes to a new script and reuse them.

**Review by jochenmanz — CHANGES_REQUESTED** [review]

**Review by jochenmanz — APPROVED** [review]

**Pull request merged**